sagemaker_processing/feature_transformation_with_sagemaker_processing_Dask/feature_transformation_with_sagemaker_processing_dask.ipynb

```diff
@@ -0,0 +1,476 @@
+    "    StandardScaler,\n",
+    ")\n",
+    "\n",
+    "warnings.filterwarnings(action=\"ignore\", category=DataConversionWarning)\n",
+    "s3_client = boto3.resource(\"s3\")\n",
+    "attempts_counter = 3\n",
+    "attempts = 0\n",
+    "\n",
+    "\n",
+    "def upload_objects(bucket, prefix, local_path):\n",
+    "    try:\n",
+    "        bucket_name = bucket  # s3 bucket name\n",
+    "        root_path = local_path  # local folder for upload\n",
+    "\n",
+    "        s3_bucket = s3_client.Bucket(bucket_name)\n",
+    "\n",
+    "        for path, subdirs, files in os.walk(root_path):\n",
+    "            for file in files:\n",
+    "                s3_bucket.upload_file(\n",
+    "                    os.path.join(path, file), prefix + \"/output/\" + file\n",
+    "                )\n",
+    "    except Exception as err:\n",
+    "        logging.exception(err)\n",
+    "\n",
+    "\n",
+    "def print_shape(df):\n",
+    "    negative_examples, positive_examples = np.bincount(df[\"income\"])\n",
+    "    print(\n",
+    "        \"Data shape: {}, {} positive examples, {} negative examples\".format(\n",
+    "            df.shape, positive_examples, negative_examples\n",
+    "        )\n",
+    "    )\n",
+    "\n",
+    "\n",
+    "if __name__ == \"__main__\":\n",
+    "    parser = argparse.ArgumentParser()\n",
+    "    parser.add_argument(\"--train-test-split-ratio\", type=float, default=0.3)\n",
+    "    args, _ = parser.parse_known_args()\n",
+    "\n",
+    "    # Get processor scrip arguments\n",
+    "    args_iter = iter(sys.argv[1:])\n",
+    "    script_args = dict(zip(args_iter, args_iter))\n",
+    "    scheduler_ip = sys.argv[-1]\n",
+    "\n",
+    "    # Start the Dask cluster client\n",
+    "    try:\n",
+    "        client = Client(\"tcp://\" + str(scheduler_ip) + \":8786\")\n",
+    "        logging.info(\"Printing cluster information: {}\".format(client))\n",
+    "    except Exception as err:\n",
+    "        logging.exception(err)\n",
+    "\n",
+    "    columns = [\n",
+    "        \"age\",\n",
+    "        \"education\",\n",
+    "        \"major industry code\",\n",
+    "        \"class of worker\",\n",
+    "        \"num persons worked for employer\",\n",
+    "        \"capital gains\",\n",
+    "        \"capital losses\",\n",
+    "        \"dividends from stocks\",\n",
+    "        \"income\",\n",
+    "    ]\n",
+    "    class_labels = [\" - 50000.\", \" 50000+.\"]\n",
+    "    input_data_path = \"s3://\" + os.path.join(\n",
+    "        script_args[\"s3_input_bucket\"],\n",
+    "        script_args[\"s3_input_key_prefix\"],\n",
+    "        \"census-income.csv\",\n",
+    "    )\n",
+    "    print(input_data_path)\n",
+    "\n",
+    "    # Creating the necessary paths to save the output files\n",
+    "    if not os.path.exists(\"/opt/ml/processing/train\"):\n",
+    "        os.makedirs(\"/opt/ml/processing/train\")\n",
+    "        os.makedirs(\"/opt/ml/processing/test\")\n",
```

Comment:
Do you also want to check if `test` exists as `makedirs` will raise `OSError` if it exists.

Reply:
Added the check for test. If it exists it should pass without raising exception